indico/modules/events/static/tasks.py

```diff
@@ -0,0 +1,109 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2015 European Organization for Nuclear Research (CERN).
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 3 of the
+# License, or (at your option) any later version.
+#
+# Indico is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Indico; if not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import unicode_literals
+
+import errno
+import os
+from datetime import timedelta
+
+from celery.schedules import crontab
+
+from indico.core.celery import celery
+from indico.core.db import db
+from indico.core.notifications import make_email, email_sender
+from indico.modules.events.static import logger
+from indico.modules.events.static.models.static import StaticSite, StaticSiteState
+from indico.util.contextManager import ContextManager
+from indico.util.date_time import now_utc
+from indico.util.fossilize import clearCache
+from indico.util.i18n import set_session_lang
+from indico.web.flask.templating import get_template_module
+from indico.web.flask.util import url_for
+
+from MaKaC.accessControl import AccessWrapper
+from MaKaC.common.offlineWebsiteCreator import OfflineEvent
+from MaKaC.webinterface.rh.conferenceBase import RHCustomizable
+
+
+@celery.task(request_context=True)
+def build_static_site(static_site):
+    static_site.state = StaticSiteState.running
+    db.session.commit()
+    clearCache()
+    try:
+        logger.info('Building static site: {}'.format(static_site))
+        set_session_lang(static_site.user.settings.get('lang'))
```

Comment:
hm.. not a big fan of this function.. i'd use `session.lang = static_site.user.settings.get('lang')` instead.